Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2523,18 +2541,67 @@ namespace Babylon
                         auto inputSourceVal = inputSourceFound->second.Value();
                         SetXRInputSourceData(inputSourceVal, inputSource);
 
-                        //inputSource already exists, find the corresponding gamepad object if enabled and set to correct values
+                        // inputSource already exists, find the corresponding gamepad object if enabled and set to correct values
                         if (inputSourceVal.Has("gamepad"))
                         {
                             auto gamepadObject = inputSourceVal.Get("gamepad").As<Napi::Object>();
                             SetXRGamepadObjectData(inputSourceVal, gamepadObject, inputSource);
                         }
                     }
+
+                    // Handle gestures. Sources that do not support a gesture will always return false
+                    bool isSelected = inputSource.GamepadObject.Buttons[0].Pressed;
+                    bool isSqueezed = inputSource.GamepadObject.Buttons[1].Pressed;
+                    if (isSelected && (m_activeSelects.find(inputSource.ID) == m_activeSelects.end()))
+                    {
+                        selectStarts.insert(inputSource.ID);
+                    }
+                    else if (!isSelected && (m_activeSelects.find(inputSource.ID) != m_activeSelects.end()))
+                    {
+                        selectEnds.insert(inputSource.ID);
+                    }
+
+                    if (isSqueezed && (m_activeSqueezes.find(inputSource.ID) == m_activeSqueezes.end()))
```

Comment:
>isSqueezed [](start = 24, length = 10)

Is it possible to be both squeezed and pressed? And if so, is it safe to fire events for both?

Reply:
It should be possible. Babylon.JS doesn't seem to use squeeze events right now, though, so nothing will break in any case.

Reply:
Thhe shell fires both when you do an air tap, by the way. There's also no documentation saying both wont be fired together.